### PR TITLE
feat: 🎸 add caching (allow eslint --cache)

### PR DESCRIPTION
**Pull request by rarmatei**
Allow the --cache option to be passed to the eslint cli engine.

Needed for: https://github.com/nrwl/nx/issues/2337

### CODE REVIEW

File: packages/builder/src/index.ts [hunk 1]
```diff
@@ -224,18 +224,16 @@ async function _lint(
...
      */
     // Give some breathing space to other promises that might be waiting.
     await Promise.resolve();
-    lintReports.push(cli.executeOnText(contents, file));
     lintedFiles.add(file);
+    filesToLint.push(file);
   }
 
-  return lintReports;
+  return [cli.executeOnFiles(filesToLint)];
```
Comment (rarmatei) [inline packages/builder/src/index.ts]: only `executeOnFiles` persists the cache to disk: https://github.com/eslint/eslint/blob/master/lib/cli-engine/cli-engine.js#L821

**Comment by JamesHenry** [comment]
Thanks a lot for this, I’ll take a look this weekend!

**Comment by JamesHenry** [comment]
I added an integration test with Angular CLI so we can be more confident about the ramifications of changes like this - triggering a rebuild

**Comment by JamesHenry** [comment]
I'm glad I did that because it has shown that this PR has changed the way data is reported.

Previously it was per file as ESLint users would expect, not it looks like it might be per TypeScript Program:

**Before** (from jest snapshot)

```sh
Linting \\"angular-cli-workspace\\"...

__ROOT__/angular-cli-workspace/src/app/app.component.ts
   2:1  error    'rxjs/Rx' import is restricted from being used. Please import directly from 'rxjs' instead                             no-restricted-imports
   5:1  error    Unexpected property on console object was called                                                                       no-restricted-syntax
  15:3  warning  Lifecycle interface 'OnInit' should be implemented for method 'ngOnInit'. (https://angular.io/styleguide#style-09-01)  @angular-eslint/use-lifecycle-interface

✖ 3 problems (2 errors, 1 warning)


__ROOT__/angular-cli-workspace/src/app/example.pipe.ts
  6:8  error  Classes decorated with @Pipe decorator should implement PipeTransform interface  @angular-eslint/use-pipe-transform-interface

✖ 1 problem (1 error, 0 warnings)


__ROOT__/angular-cli-workspace/src/app/example.component.ts
  12:3  error  Use @Input rather than the \`inputs\` metadata property (https://angular.io/styleguide#style-05-12)    @angular-eslint/no-inputs-metadata-property
  13:3  error  Use @Output rather than the \`outputs\` metadata property (https://angular.io/styleguide#style-05-12)  @angular-eslint/no-outputs-metadata-property
  14:3  error  Use @Output rather than the \`host\` metadata property (https://angular.io/styleguide#style-05-12)     @angular-eslint/no-host-metadata-property
  18:3  error  In the class \\"ExampleComponent\\", the output property \\"onFoo\\" should not be prefixed with on          @angular-eslint/no-output-on-prefix

✖ 4 problems (4 errors, 0 warnings)


__ROOT__/angular-cli-workspace/src/app/app.component.spec.ts
  19:6  error  Strings must use singlequote  quotes

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the \`--fix\` option.
```

**After** (From terminal output locally)

```sh

/Users/james/Documents/open-source/angular-eslint/packages/integration-tests/fixtures/angular-cli-workspace/src/app/app.component.ts
   2:1  error    'rxjs/Rx' import is restricted from being used. Please import directly from 'rxjs' instead                             no-restricted-imports
   5:1  error    Unexpected property on console object was called                                                                       no-restricted-syntax
  15:3  warning  Lifecycle interface 'OnInit' should be implemented for method 'ngOnInit'. (https://angular.io/styleguide#style-09-01)  @angular-eslint/use-lifecycle-in
terface

/Users/james/Documents/open-source/angular-eslint/packages/integration-tests/fixtures/angular-cli-workspace/src/app/example.pipe.ts
  6:8  error  Classes decorated with @Pipe decorator should implement PipeTransform interface  @angular-eslint/use-pipe-transform-interface

/Users/james/Documents/open-source/angular-eslint/packages/integration-tests/fixtures/angular-cli-workspace/src/app/example.component.ts
  12:3  error  Use @Input rather than the `inputs` metadata property (https://angular.io/styleguide#style-05-12)    @angular-eslint/no-inputs-metadata-property
  13:3  error  Use @Output rather than the `outputs` metadata property (https://angular.io/styleguide#style-05-12)  @angular-eslint/no-outputs-metadata-property
  14:3  error  Use @Output rather than the `host` metadata property (https://angular.io/styleguide#style-05-12)     @angular-eslint/no-host-metadata-property
  18:3  error  In the class "ExampleComponent", the output property "onFoo" should not be prefixed with on          @angular-eslint/no-output-on-prefix

✖ 8 problems (7 errors, 1 warning)


/Users/james/Documents/open-source/angular-eslint/packages/integration-tests/fixtures/angular-cli-workspace/src/app/app.component.spec.ts
  19:6  error  Strings must use singlequote  quotes

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by rarmatei** [comment]
Hi James! Oh that's a great find! I rebased and fixed the issue, and the integration tests seem to pass now. Thanks again for the review!

**Pull request merged**